tests/framework/BaseYiiTest.php

```diff
@@ -60,4 +62,26 @@ public function testPowered()
     {
         $this->assertTrue(is_string(Yii::powered()));
     }
+
+    public function testCreateObjectCallable()
+    {
+        Yii::$container = new Container();
+
+        // Test passing in of normal params combined with DI params.
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a) {
+            return $a === 'a';
+        }, ['a']));
+
+
+        $singer = new Singer();
+        $singer->firstName = 'Bob';
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a) {
+            return $singer->firstName === 'Bob';
+        }, [$singer, 'a']));
+
+
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a = 3) {
+            return true;
+        }));
```

Comment:
I'm testing something else than you are... I'm testing if it will correctly use a class type parameter instead of instantiating a new one. You're testing if `createObject`  will return the result of the closure.